etc/kayobe/ansible/dashboard.overview.j2

```diff
@@ -0,0 +1,314 @@
+{% set counter = { 'counter1': 0, 'counter2': 1 } %}
+{% macro increment1(dct, key, inc=8)%}
+    {% if dct.update({key: dct[key] + inc}) %} {% endif %}
+{% endmacro %}
+{% macro increment2(dct, key, inc=1)%}
+    {% if dct.update({key: dct[key] + inc}) %} {% endif %}
+{% endmacro %}
+from grafanalib.core import *
+
+dashboard = Dashboard(
+    title="Openstack Overview Dashboard",
+    description="",
+    timezone="browser",
+    templating=Templating([
+        Template(
+            type="datasource",
+            dataSource="default",
+            name="datasource",
+            label="Prometheus",
+            query="prometheus"
+        ),
+        Template(
+            type="query",
+            name="node",
+            dataSource='$datasource',
+            label="node",
+            query={ "query": "label_values(node_cpu_seconds_total, instance)",
+                    "refId": "StandardVariableQuery"
+            },
+            refresh=5
+        ),
+       ],
+     ),
+     panels=[
+        RowPanel(
+        title = "Service Stats",
+        collapsed=False,
+        gridPos=GridPos(h=1, w=24, x=0, y=0),
+        ),
+        {% for service in openstack_enabled_services %}
+        Stat(
+                    title="{{ service.name }}",
+                    dataSource="$datasource",
+                    orientation="horizontal",
+                    mappings=[
+                        StatValueMappings(
+			StatValueMappingItem('DOWN', '0' , 'red', 0),
+			StatValueMappingItem('WARN', '1' , 'orange', 1),
+			StatValueMappingItem('WARN', '2' , 'orange', 2),
+			StatValueMappingItem('UP', '3' , 'green', 3),
+                       ),
+                        StatRangeMappings(text='UP', color='green', index=4, startValue=3, endValue=20)
+                     ],
+                     targets=[
+                     Target(
+                            expr='sum(haproxy_backend_up{backend="{{ service.backend }}"})',
+                            datasource="$datasource",
+                        ),
+                    ],
+                    gridPos=GridPos(h=3, w=8, x={{ counter.counter1}}, y={{ counter.counter2 }}),
+                    {{ increment1(counter, 'counter1') }}
+                    {% if (counter.counter1 == 24) %}
```

Comment:
Does this depend on screen resolution? It would be nice if we could somehow rely on auto-layout, I guess that's not possible?

Reply:
It does not depend on screen resolution. This sets the position of the panel within the row. if X is a smaller value than the previous panel it will place it underneath. The counter resets at 24, but depends on the size of the H and W of the panel. a Smaller panel will require a different value.